### ISSUE TITLE
Using a "relation" widget as "field" inside of a "list" widget causes problems.

### ISSUE THREAD

**Issue by richtera** [netlify/netlify-cms#1670]
**Describe the bug**
When using a relation widget inside of a list widget as a single field it causes problems. You can type one character but you can't delete it or find any related items. It also doesn't look right:

**To Reproduce**
Inside of one of the collections in your config.yml

This works:
```yaml
      - label: 'Tags'
        name: 'tags'
        widget: 'list'
        field:
          - label: 'Tag'
            widget: 'string'
```
This causes the problem:
```yaml
      - label: 'Activities'
        name: 'activities'
        widget: 'list'
        field:
          - label: 'Activity'
            name: 'activity'
            widget: 'relation'
            collection: 'activity'
            displayFields:
              - "slug"
              - "title"
            searchFields:
              - "slug"
              - "title"
            valueField: 'slug'
```

**Expected behavior**
Using a list of relation items should work.

**Screenshots**
![image](https://user-images.githubusercontent.com/708186/44613169-f1bb6000-a7dd-11e8-8bed-3cb3dec3916f.png)

The debugger shows this:
![image](https://user-images.githubusercontent.com/708186/44613199-6a222100-a7de-11e8-82bd-07631e7662e5.png)
Sorry about not having symbols inside of the cms code  (I couldn't figure out how to merge the module-resolver babel plugin to allow me to use the sources directly)

**Applicable Versions:**
 - Netlify CMS version: 1.9.3 (I have not re-tried the mono repo)
 - Git provider: github
 - OS: Mac OS
 - Browser version Chrome 68.0.3440.106 (Official Build) (64-bit)

 - Node.JS version: 9.11.1

**CMS configuration**
```

# auto - insert backend config
# <BACKEND_CONFIG>

media_folder: assets/images
public_folder: /static/images
publish_mode: editorial_workflow

collections:
  - name: "page"
    label: "Pages"
    folder: "content/pages"
    create: true
    format: yaml-frontmatter
    extension: md
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Title", name: "title", widget: "string", required: false}
      - {label: "Slug", name: "slug", widget: "string", required: true}
      - {label: "Content", name: "body", widget: "markdown"}
      - {label: "Tags", name: "tags", widget: "list", required: false}

  - name: "category"
    label: "Categories"
    folder: "content/categories"
    create: true
    format: yaml-frontmatter
    extension: md
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "ID", name: "id", widget: "string", required: true}
      - {label: "Title", name: "title", widget: "string", required: true}
      - {label: "Tags", name: "tags", widget: "list", required: false}

  - name: "activity"
    label: "Activities"
    folder: "content/activities"
    create: true
    format: yaml-frontmatter
    extension: md
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Title", name: "title", widget: "string", required: false}
      - {label: "Slug", name: "slug", widget: "string", required: true}
      - {label: "Category", name: "category", widget: "relation", collection: 'category', searchFields: ["id", "title"], displayFields: ["id", "title"], valueField: "id"}
      - {label: "Content", name: "body", widget: "markdown"}
      - {label: "Tags", name: "tags", widget: "list", required: false}

  - name: "unit"
    label: "Unit"
    folder: "content/units"
    create: true
    format: yaml-frontmatter
    extension: md
    delete: false
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "ID", name: "id", widget: "string", required: false}
      - {label: "Code Number", name: "code", widget: "string"}
      - {label: "Active (display and Stripe)", name: "active", widget: "boolean", default: true}
      - {label: "Title (display and Stripe)", name: "title", widget: "string", required: false}
      - {label: "Publish Date", name: "date", widget: "datetime"}
      - {label: "Tags", name: "tags", widget: "list", required: false}
      - {label: 'Price (for display)', name: 'price', widget: "text"}
      - {label: 'Amount (for Stripe in cents)', name: 'amount', widget: "number", valueType: "int", min: 0}
      - {label: 'Currency (for Stripe)', name: 'currency', default: 'USD', widget: 'text'}
      - label: "Interval"
        name: "interval"
        widget: 'select'
        options:
          - label: Day
            value: day
          - label: Week
            value: week
          - label: Month
            value: month
          - label: Year
            value: year
      - {label: 'Interval Count', name: 'interval_count', widget: "number", valueType: "int", min: 0}
      - {label: 'Auto Cancel (during purchase)', name: 'autoCancel', widget: "boolean", default: true}
      - {label: 'Order', name: 'order', widget: "number"}
      - {label: 'Only Offer', name: 'offer', widget: "boolean", required: false, default: false}
      - label: 'Tags'
        name: 'tags'
        widget: 'list'
        field:
          - label: 'Tag'
            widget: 'string'
      - label: "Randomize Activites"
        name: 'randomActivities'
        widget: 'boolean'
        default: true
        required: false
      - label: 'Activities'
        name: 'activities'
        widget: 'list'
        field:
          - label: 'Activity'
            name: 'activity'
            widget: 'relation'
            collection: 'activity'
            displayFields:
              - "slug"
              - "title"
            searchFields:
              - "slug"
              - "title"
            valueField: 'slug'
      - {label: 'Subscriptions', name: 'subscriptions', widget: "list", field: {label: 'Subscription', widget: 'string'}}
      - {label: 'Survey', name: 'survey', widget: 'relation', collection: 'activity', displayFields: ['slug', 'title'], searchFields: ['slug', 'title'], valueField: 'slug'}
      - {label: "Slug", name: "slug", widget: "string", required: true}
```

**Additional context**
Sorry that I am stuck on the non-mono-repo version since I have not figured out how to do a live patch of that version.

**Comment by tech4him1** [comment]
That bug was likely fixed in #1395, released in 2.0.10. 

What problems are you having using the mono-repo version?

**Comment by erquhart** [comment]
Confirmed fixed in 2.0.10.

**Comment by richtera** [comment]
Awesome, thank you. I will try to merge 2.0.10 into my code. @tech4him1 The monorepo is preventing me from installing npm patches inside of netlify deploy from a git repo. I need to investigate patching the whole mono repo and creating an npm org and installing the modules from a scoped repo.
